gen/internals.py

```diff
@@ -246,44 +246,62 @@ def __eq__(self, other):
 
 
 class Source:
-    """ Object to define how arguments can be specified, including argument
-    calculation, default values, validation, and any conditional variations
-    of those three
-    """
+    """Defines config arguments, including calculation, defaults, validation, and whether arguments are secrets."""
     def __init__(self, entry=None, is_user=False):
         """ Entry is a dict of the following form:
         {
           'validate': [validate_my_arg_fn],
           'default': {arg: value},  # may be overridden by user args
-          'must': {arg: value}  # may NOT be overridden by user args
+          'must': {arg: value},  # may NOT be overridden by user args
+          'secret': [arg],
           'conditional': {arg: {val_1: add_entry_1, val_2: add_entry_2}}
         }
         validate_my_arg_fn: a function that will be called against the argument names it uses
         add_entry_*: additional entries like the object above
         """
         self.setters = dict()
         self.validate = list()
+        self.secret = list()
         self.is_user = is_user
         if entry:
             self.add_entry(entry, False)
 
     def add_setter(self, name, value, is_optional, conditions):
-        self.setters.setdefault(name, list()).append(Setter(name, value, is_optional, conditions, self.is_user))
+        self.setters.setdefault(name, list()).append(
+            Setter(name, value, is_optional, conditions, self.is_user)
+        )
 
     def add_conditional_scope(self, scope, conditions):
         # TODO(cmaloney): 'defaults' are the same as 'can' and 'must' is identical to 'arguments' except
         # that one takes functions and one takes strings. Simplify to just 'can', 'must'.
-        assert scope.keys() <= {'validate', 'default', 'must', 'conditional'}
+        assert scope.keys() <= {'validate', 'default', 'must', 'secret', 'conditional'}
```

Comment:
`.issubset` would be more intuitive (IMO).

Reply:
Eh this syntax doesn't really bug me. And I'm just trying to add a key to the allowed set here.

Reply:
The review comment was not for your change. :) It was the existing use of tricky syntax.